src/values/instruction_value.rs

```diff
@@ -141,6 +141,20 @@ impl<'ctx> InstructionValue<'ctx> {
         }
     }
 
+    /// Get name of the `InstructionValue`.
+    pub fn get_name(&self) -> &CStr {
+        self.instruction_value.get_name()
```

Comment:
I suspect this might return a null ptr if the instruction is a void type

Reply:
I tried to check if a `&CStr` is a null pointer, but it seems a null ptr is also a valid `&CStr` type?

What I'm currently doing is to implement another function:

```
    /// Get name of the `InstructionValue` or return a default name.
    pub fn get_name_or_default(&self) -> String {
        match self.get_name().to_str() {
            Ok(name) => name.to_string(),
            _ => "<empty-instruction-name>".to_string(),
        }
    }
```

How about we change the output of the existing function `get_name` to `Option<String>`, like below?

```
    pub fn get_name(&self) -> Option<String> {
         match self.instruction_value.get_name().to_str() {
            Ok(name) => Some(name)
            _ => None
        }
    }
```

In this case, we don't need to care about the instruction type.

Reply:
We should return an `Option<&CStr>`, that way making an allocation is up to the end user not inkwell

Reply:
All fixed as suggested!